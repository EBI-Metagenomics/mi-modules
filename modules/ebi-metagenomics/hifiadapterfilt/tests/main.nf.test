nextflow_process {

    name "Test Process HIFIADAPTERFILT"
    script "../main.nf"
    process "HIFIADAPTERFILT"

    tag "modules"
    tag "modules_ebimetagenomics"
    tag "hifiadapterfilt"

    test("hifiadapterfilt - fastq.gz") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // TODO meta map
                    file("${moduleDir}/tests/fixtures/SRR30463389_subreads.fastq.gz", checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("hifiadapterfilt - fastq.gz - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // TODO meta map
                    file("${moduleDir}/tests/fixtures/SRR30463389_subreads.fastq.gz", checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
