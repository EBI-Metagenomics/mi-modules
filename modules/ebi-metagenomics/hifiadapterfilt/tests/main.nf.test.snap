{
    "hifiadapterfilt - fastq.gz": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "blast_search": [
                    
                ],
                "filt": [
                    
                ],
                "headers": [
                    
                ],
                "stats": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-17T12:34:18.179022867"
    },
    "hifiadapterfilt - fastq.gz - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "blast_search": [
                    
                ],
                "filt": [
                    
                ],
                "headers": [
                    
                ],
                "stats": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-17T12:34:23.562113107"
    }
}